boot.php

```diff
@@ -785,60 +785,85 @@ function get_scheme() {
 		return($this->scheme);
 	}
 
+	/**
+	 * @brief Retrieves the Friendica instance base URL
+	 *
+	 * Caches both SSL and non-SSL version for performance
+	 *
+	 * @param bool $ssl
+	 * @return string
+	 */
 	function get_baseurl($ssl = false) {
 
 		// Is the function called statically?
-		if (!is_object($this))
-			return(self::$a->get_baseurl($ssl));
+		if (!is_object($this)) {
+			return self::$a->get_baseurl($ssl);
+		}
 
-		$scheme = $this->scheme;
+		if (!isset($this->baseurl[$ssl ? 'https' : 'http'])) {
```

Comment:
I'm still not totally convinced. The parameter doesn't really distinguish between "http" and "https". So maybe it would be better to better use ```$cache_index = $ssl ? 'true' : 'false';```` and to add some documentation that it is done this way since array keys mustn't be boolean.